### PR TITLE
extracting resource.url from lessons

### PR DESCRIPTION
**Pull request by chrisdugne**
getting resource.url properly for jwplayer
requires https://github.com/CoorpAcademy/components/pull/1513

### Result and observation

- [ ] **Breaking changes ?**  
       If checked, what have you broken ?

- [ ] **Extra dependency ?**
      If checked, Which extra dependency did you add ? (name, purpose, link ...).

### Testing Strategy

- [x] Already covered by tests
- [x] Manual testing
- [ ] Unit testing

### CODE REVIEW

**Comment by chrisdugne** [comment]
[Mobile - bug : l'app saute quand je clique sur l'onglet ressource](https://trello.com/c/RZGMNQnD/1158-mobile-bug-lapp-saute-quand-je-clique-sur-longlet-ressource)

File: src/components/lesson.stories.js [hunk 1]
```diff
@@ -14,7 +16,8 @@ const videoSubtitles = createVideo({
   description: 'Foo bar baz - Video subtitles',
   subtitleRef: 'foobarbaz'
 });
-const resources = [video, videoSubtitles, pdf];
+const lessons = [video, videoSubtitles, pdf];
+const resources: Array<Resource> = reduceToResources(lessons);
```
Comment (alanlanglois) [inline src/components/lesson.stories.js]: nice 👍

File: src/layer/data/_types.js [hunk 2]
```diff
@@ -8,16 +8,10 @@ import type {
   ChapterRule
 } from '@coorpacademy/player-services';
 
-import type {Context, Lesson} from '@coorpacademy/progression-engine';
+import type {Context, Lesson, ResourceMimeType} from '@coorpacademy/progression-engine';
```
Comment (alanlanglois) [inline src/layer/data/_types.js]: 👍

File: src/modules/uri.js [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
 export const getCleanUri = (originalUri: string): string =>
   originalUri.replace(/(http:|https:|)\/\//g, 'https://');
+
+export const getResourceUrl = (resource: Lesson): string | void => {
+  const videoId = resource && resource.videoId;
+  const mediaUrl = resource && resource.mediaUrl;
+  const downloadUrl = resource && resource.downloadUrl;
+  const url = (resource.type === RESOURCE_TYPE.VIDEO && downloadUrl) || mediaUrl;
+
+  const jwpWithNoUrl = !url || resource.mimeType === 'application/jwplayer';
```
Comment (alanlanglois) [inline src/modules/uri.js]: Wouldn't be better to add it to the type 'ResourceMimeType'?

**Pull request merged**